oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/token/JwtAccessTokenOAuth2TokenIssuer.java

```diff
@@ -0,0 +1,144 @@
+	 *
+	 * @param validators new validators collection
+	 */
+	public void setValidators(Iterable<JwtOAuth2TokenValidator> validators) {
+		this.validators = validators;
+	}
+
+	/**
+	 * Set new token customizer. Customizer is applied before issuing the token and it is a best extension point to implement
+	 * specific token generation logic.
+	 *
+	 * @param customizer new customizer
+	 */
+	public void setCustomizer(JwtOAuth2TokenCustomizer customizer) {
+		this.customizer = customizer;
+	}
+
+	/**
+	 * Issue a new Jwt access token. The generated token's claim names are compliant with RFC7519.
+	 *
+	 * @see OAuth2AccessToken
+	 * @see Jwt
+	 * @see <a href="https://tools.ietf.org/html/rfc7519#section-4">RFC7519 section 4</a>
+	 *
+	 * @param tokenRequest the request
+	 * @return a new OAuth2Token with metadata
+	 */
+	@Override
+	public OAuth2TokenResult<OAuth2AccessToken> issue(@NonNull OAuth2TokenRequest tokenRequest) {
+		Assert.notNull(tokenRequest.getPrincipalName(), "resourceOwner cannot be null");
+		Assert.notNull(tokenRequest.getRegisteredClient(), "registeredClient cannot be null");
+		Assert.notNull(tokenRequest.getTimeToLive(), "timeToLive cannot be null");
+
+		Instant issuedAt = Instant.now(this.clock);
+		Instant expiresAt = issuedAt.plus(tokenRequest.getTimeToLive());
+
+		JoseHeader.Builder joseHeaderBuilder = JoseHeader.withAlgorithm(SignatureAlgorithm.RS256);
+		Set<String> scopes = tokenRequest.getScopes();
+		JwtClaimsSet.Builder jwtClaimsSetBuilder = JwtClaimsSet.withClaims()
+				.subject(tokenRequest.getPrincipalName())
+				.audience(Collections.singletonList(tokenRequest.getRegisteredClient().getClientId()))
+				.issuedAt(issuedAt)
+				.expiresAt(expiresAt)
+				.notBefore(issuedAt)
+				.claim(OAuth2ParameterNames.SCOPE, scopes);
+
+		this.customizer.customize(joseHeaderBuilder, jwtClaimsSetBuilder, tokenRequest);
+
+		JoseHeader joseHeader = joseHeaderBuilder.build();
+		JwtClaimsSet claimsSet = jwtClaimsSetBuilder.build();
+		this.validators.forEach(jwtOAuth2TokenValidator -> jwtOAuth2TokenValidator.validate(joseHeader, claimsSet, tokenRequest));
+
+		Jwt jwt = jwtEncoder.encode(joseHeaderBuilder.build(), jwtClaimsSetBuilder.build());
+		OAuth2AccessToken token = new OAuth2AccessToken(OAuth2AccessToken.TokenType.BEARER, jwt.getTokenValue(), issuedAt, expiresAt, scopes);
+		OAuth2TokenMetadata metadata = OAuth2TokenMetadata
+				.builder()
+				.metadata(OAuth2TokenMetadata.JWT, jwt)
+				.metadata(OAuth2TokenMetadata.AUTHORIZED_SCOPES, scopes)
+				.build();
+
+		return OAuth2TokenResult.of(token, metadata);
+	}
+
+	private List<JwtOAuth2TokenValidator> getDefaultValidators() {
+		return Collections.singletonList(new TokenLifetimeValidator());
+	}
+
+	/**
+	 * Default token lifetime validator, ensures that token's expiresAt is set and after issuedAt
+	 */
+	private static class TokenLifetimeValidator implements JwtOAuth2TokenValidator {
```

Comment:
Are you able to reuse `JwtTimestampValidator` and/or `JwtClaimValidator`?

Reply:
See below, I don't think these validators needed at all.